cloudify_agent/operations.py

```diff
@@ -208,27 +184,57 @@ def create_new_agent_dict(old_agent):
     new_agent['manager_file_server_url'] = get_manager_file_server_url()
     # Assuming that if there is no version info in the agent then
     # this agent was installed by current manager.
-    new_agent['old_agent_version'] = old_agent.get('version', _VERSION)
+    new_agent['old_agent_version'] = old_agent['version']
     return new_agent
 
 
+@contextmanager
+def _celery_client(ctx, agent):
+    # We retrieve broker url from old agent in order to support
+    # cases when old agent is not connected to current rabbit server.
+    if 'broker_config' in agent:
+        broker_config = agent['broker_config']
+    else:
+        broker_config = ctx.bootstrap_context.broker_config()
+    broker_url = utils.internal.get_broker_url(broker_config)
+    ctx.logger.info('Connecting to {0}'.format(broker_url))
+    celery_client = celery.Celery()
+    # We can't pass broker_url to Celery constructor because it would
+    # be overriden by the value from broker_config.py.
+    config = {
+        'BROKER_URL': broker_url,
+        'CELERY_RESULT_BACKEND': broker_url
+    }
+    if agent['version'] != '3.2':
+        config['CELERY_TASK_RESULT_EXPIRES'] = \
+            defaults.CELERY_TASK_RESULT_EXPIRES
+    fd, cert_path = tempfile.mkstemp()
+    os.close(fd)
+    try:
+        if broker_config.get('broker_ssl_enabled'):
+            with open(cert_path, 'w') as cert_file:
+                cert_file.write(broker_config.get('broker_ssl_cert', ''))
+            broker_ssl = {
+                'ca_certs': cert_path,
+                'cert_reqs': ssl.CERT_REQUIRED
+            }
+        else:
+            broker_ssl = False
+        config['BROKER_USE_SSL'] = broker_ssl
+        celery_client.conf.update(**config)
+        yield celery_client
+    finally:
+        os.remove(cert_path)
+
+
 def create_agent_from_old_agent(install_agent_timeout=300):
     if 'cloudify_agent' not in ctx.instance.runtime_properties:
         raise NonRecoverableError(
             'cloudify_agent key not available in runtime_properties')
     old_agent = ctx.instance.runtime_properties['cloudify_agent']
+    old_agent['version'] = old_agent.get('version', _VERSION)
```

Comment:
Can we use a mechanism similar to
https://github.com/cloudify-cosmo/cloudify-manager/pull/365/files#diff-65e4c2d1d79b4b1b3b988c9be1803305R306
instead of hard coding the default `_VERSION` value?